docs/design/design.md

```diff
@@ -0,0 +1,102 @@
+
+	+ A tracker and scheduler in the P2P network that chooses appropriate downloading net-path for each peer.
+	+ A CDN server that caches downloaded data from source to avoid downloading same files repeatedly.
+
+2. Dfclient, Dfclient contains two components:
+
+	+ Dfget, Dfget is the Dragonfly client used for downloading files. It's similar to wget. Meanwhile, it also plays the role of a peer with uploader, which can transfer data between each other in the P2P network.
+	+ Dfdaemon, Dfdaemon is only used for pulling images. It establishes a proxy between dockerd/pouchd and registry.Dfdaemon filters out layer fetching requests from all requests sent by dockerd/pouchd when pulling images, then uses dfget to download these layers.
+
+As mentioned above, we use the following picture to make it easier for you to understand.
+
+![architecture.png](../images/design/architecture.png)
+
+## Workflows
+
+### Dfget and Supernode's communication workflows
+
+When Dfget try to download files/images from Supernode, there are three steps to follow:
+
+1. Register
+2. Download
+3. ServiceDown
+
+Below is a diagram illustrating that how dfget download a file from Supernode.
+
+![dfclient_supernode.png](../images/design/dfclient_supernode.png)
+
+#### Register
+
+1. Dfget should launch a server as a peer and refister to Supernode.
+2. Dfget should tell the Supernode some info about files/images to be downloaded.
+
+#### Download
+
+The Dfget download is not a one-time work, it will download a piece at a time with a loop until the download is complete. It is mainly divided into the following three steps:
+
+1. Get peers information with the pieces to be downloaded.
+2. Download the pieces from the specified peers.
+3. Update the pieceInfo to tell supernode that it has downloaded the piece for two things:
+	+ Provides a download service for other nodes in the network that need the piece
+	+ Get the information about other pieces that need to be downloaded
+
+#### ServiceDown
+
+When Dfget lasts in five minutes without downloading or uploading task, uploader of dfget will automatically sends a serviceDown request to supernode.
+
+### Supernode internal component workflows
+
+Based on the API and the function of the system, we have five modules so far as follows:
+
++ TaskMgr
++ PeerMgr
++ DfgetTaskMgr
++ CDNMgr
++ SchedulerMgr
+
+Based on the above definitions, we can easily guess that each component is responsible for manipulating the corresponding resource object with component name. However, we have to have a mention about SchedulerMgr, the SchedulerMgr has two main responsibilities:
+
++ Responsible for maintaining the correspondence between peer and pieces.
++ Responsible for calculating scheduling results according to certain rules.
+
+We will show the Supernode internal component workflows by three workflows mentioned above.
+
+Register
+
+![register.png](../images/design/register.png)
```

Comment:
Those two pictures look like a little obscure. Can you switch more clear pictures?